### DISCUSSION TITLE
How can I download Language Server executable?

### DISCUSSION THREAD

**Discussion by lawrence-laz** [dotnet/roslyn#69731] [Q&A]
I initially expected the language server to be bundled together with .net sdk, but it doesn't seem to be there.

After looking at the [project file](https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Microsoft.CodeAnalysis.LanguageServer.csproj) I see that the project gets bundled as a NuGet package, but it's not published.

How do I download the language server?

**Comment by 616b2f** [comment]
I am also interested in the same thing, can the Language Server nuget be published to the public nuget.org registry? This would make it easier for other editors (like Neovim) to be able to actually pull and use it.

**Comment by hez2010** [comment]
Access the roslyn CI at https://dev.azure.com/dnceng-public/public/_build?definitionId=95
Pick the latest successful run, then you will be able to access the artifacts by clicking the "## published" button
![image](https://github.com/user-attachments/assets/64dc8aa5-81b0-4d6e-afba-9646202b073f)

The language server package is placed in "Bootstrap Packages - Default", download it to get the language server nupkg.

**Comment by dibarbet** [comment]
We do publish a nuget package (not to nuget.org though).  See https://github.com/dotnet/roslyn/issues/71474#issuecomment-2177303207 for details.

**Reply by 616b2f** [reply]
Is there any chance we could publish it as a nuget tool (dotnet tool) to nuget.org?

Because it's hard to use and hard to download with dotnet cli tool and otherwise also hard to setup with other editors then vs code.
